src/data/multiset/gcd.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Aaron Anderson
+-/
+import data.multiset.lattice
+import algebra.gcd_monoid
+
+/-!
+# GCD and LCM operations on multisets
+
+TODO: Replace all of this with a tactic and `data.multiset.lattice`
+-/
+
+namespace multiset
+variables {α : Type*} [comm_cancel_monoid_with_zero α] [nontrivial α] [gcd_monoid α]
+
+/-! ### lcm -/
+section lcm
+
+/-- Least common multiple of a multiset -/
+def lcm (s : multiset α) : α := s.fold gcd_monoid.lcm 1
+
+@[simp] lemma lcm_zero : (0 : multiset α).lcm = 1 :=
+fold_zero _ _
+
+@[simp] lemma lcm_cons (a : α) (s : multiset α) :
+  (a :: s).lcm = gcd_monoid.lcm a s.lcm :=
+fold_cons_left _ _ _ _
+
+@[simp] lemma lcm_singleton {a : α} : (a::0).lcm = normalize a := by simp
+
+@[simp] lemma lcm_add (s₁ s₂ : multiset α) : (s₁ + s₂).lcm = gcd_monoid.lcm s₁.lcm s₂.lcm :=
+eq.trans (by simp [lcm]) (fold_add _ _ _ _ _)
+
+lemma lcm_dvd {s : multiset α} {a : α} : s.lcm ∣ a ↔ (∀b ∈ s, b ∣ a) :=
+multiset.induction_on s (by simp)
+  (by simp [or_imp_distrib, forall_and_distrib, lcm_dvd_iff] {contextual := tt})
+
+lemma dvd_lcm {s : multiset α} {a : α} (h : a ∈ s) : a ∣ s.lcm :=
+lcm_dvd.1 (dvd_refl _) _ h
+
+lemma lcm_mono {s₁ s₂ : multiset α} (h : s₁ ⊆ s₂) : s₁.lcm ∣ s₂.lcm :=
+lcm_dvd.2 $ assume b hb, dvd_lcm (h hb)
+
+@[simp] lemma normalize_lcm (s : multiset α) : normalize (s.lcm) = s.lcm :=
+multiset.induction_on s (by simp) $ λ a s IH, by simp
+
+variables [decidable_eq α]
+
+@[simp] lemma lcm_erase_dup (s : multiset α) : (erase_dup s).lcm = s.lcm :=
```

Comment:
Can you use `multiset.fold_erase_dup_idem` to prove this?

Reply:
Same as above